packages/compass-app-stores/src/stores/instance-store.ts

```diff
@@ -323,7 +323,15 @@ export function createInstancesStore(
         }
       );
 
-      on(globalAppRegistry, 'view-created', maybeAddAndRefreshCollectionModel);
+      on(
+        globalAppRegistry,
+        'view-created',
+        (ns: string, { connectionId }: { connectionId?: string } = {}) => {
+          if (connectionId === instanceConnectionId) {
```

Comment:
Maybe I'm misremembering (and I admittedly haven't followed each detail of the project), but weren't we supposed to have helpers that specifically avoid the problem of listening to events on `globalAppRegistry`, then having to filter for the specific connection we're interested in?

(If I'm misremembering: That's fine :slightly_smiling_face: If I'm not, and this is a TODO, would it be possible to add a reference to the relevant ticket?)

Reply:
IIRC we purposely kept aside the idea of having the possibility of listening via this ConnectionScopedAppRegistry and limited that only to acts as event forwarder. But there is indeed a ticket (COMPASS-7815) in this project that aims to remove the dependency on globalAppRegistry events in compass-app-stores.